### PR TITLE
Input refactor

### PR DESCRIPTION
**Pull request by TrueDoctor**
This PR reworks the input system to be more distributed and remove redundancy 

---
This change is [Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)

### CODE REVIEW

**Comment by Keavon** [comment]
Diagram

**Review by TrueDoctor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)* status: 6 of 26 files reviewed, 11 unresolved discussions (waiting on @Keavon, @pkupper, @T0mstone, and @TrueDoctor)

___
*[core/editor/src/document.rs, line 9 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M__pYj16bEjj8g6vwOn:-M_avV1iEg_tD7XhflNc:buo140v) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/document.rs#L9)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Should the name not be part of document_core::document::Document instead?
</blockquote></details>

I'm not sure the name is something inherent to the bytes you are saving, I think it would be only associated with the filename which the document library will most likely not handle
___
*[core/editor/src/lib.rs, line 24 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M__q5-jAuxYbXWiQwgt:-M_avpwY17fqZRy8xmfJ:b-z04bzp) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/lib.rs#L24)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

What makes this an SvgDocument? Just because it is rendered to SVG at the moment? And why would it need to be reexported in Editor?
</blockquote></details>

We for sure can get rid of the doc inline.  SvgDocument because it currently is and always will be an svg document. The pub reexport ist mostly there for convenience will be either made private or removed in the future
___
*[core/editor/src/dispatcher/actions.rs, line 43 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M__rRhYD8O70hzjhkOW:-M_awUR-09NEUn_5ohRR:bnciu4s) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/actions.rs#L43)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Rename to PointerMove to accommodate future input methods (e.g. Pen Tablet, Touchscreen)
</blockquote></details>

I'm not sure if those would be the same input? Would not some devices have a pressure value of sorts? Also the js currently onlys sends mouse move events but this can be added in a separate pr
___
*[core/editor/src/dispatcher/document_event_handler.rs, line 69 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_a3z2hEjkDUIHv1xPa:-M_axKeRENl6Z6IZIkQ-:bt7uoh9) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/document_event_handler.rs#L69)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

replace while with Vec<T>::retain to get rid of index.
</blockquote></details>

That would be less efficient because you would have to iterate the vec twice.
This code is pretty much adopted form the filter_drain proposal
___
*[core/editor/src/dispatcher/events.rs, line 40 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_aphT8-LLuLH4vSZHf:-M_axfaWClNSq6Laz6PP:bmm3t8s) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/events.rs#L40)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Why would there be a need for an Action event?
</blockquote></details>

This would be for something like clicking a ui button in the frontend
___
*[core/editor/src/dispatcher/input_manager.rs, line 11 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_aqGIbCKowaSptQSGD:-M_ayB2pEGapSAcmDujI:bqdd92v) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/input_manager.rs#L11)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Rename to "pressed" as it is more common and doesn't have a second meaning
</blockquote></details>

adjective: pressed
forced to enlist in the army or navy.

2.
flattened, shaped, or smoothed by the application of pressure, typically from an iron.
"immaculately pressed trousers"

There is a slight semantic difference between expressing that a key is currently in the held down state, vs. the key having been pressed.
If I press a key and let go of it, it is still pressed but no longer depressed

File: are_we_proc_macro [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Hell Yeah - TrueDoctor, 2021
+Hell Yeah - T0mstone, 2021
```
Comment (T0mstone) [inline are_we_proc_macro:2]: This is not a serious addition, it expresses how much we rejoiced when our respective proc macros worked

Comment (Keavon) [reply are_we_proc_macro:2]: 😄

File: core/editor/src/communication/dispatcher.rs [hunk 2]
```diff
@@ -39,9 +40,11 @@ impl Dispatcher {
 			InputPreprocessor(message) => self.input_preprocessor.process_action(message, (), &mut self.messages),
 			InputMapper(message) => self.input_mapper.process_action(message, &self.input_preprocessor, &mut self.messages),
 		}
-		let message = self.messages.drain(..1).next();
-		if let Some(message) = message {
-			self.handle_message(message)?;
+		if !self.messages.is_empty() {
+			let message = self.messages.drain(..1).next();
+			if let Some(message) = message {
```
Comment (T0mstone) [inline core/editor/src/communication/dispatcher.rs:45]: Isn't this redundant? If `self.messages` isn't empty, then `drain(..1).next()` will always return `Some`, won't it?

Comment (TrueDoctor) [reply core/editor/src/communication/dispatcher.rs:45]: It panics, further cleanup required

Comment (T0mstone) [reply core/editor/src/communication/dispatcher.rs:45]: Sorry but I don't get what you're saying

Comment (TrueDoctor) [reply core/editor/src/communication/dispatcher.rs:45]: Drain panics if the vec is empty

Comment (T0mstone) [reply core/editor/src/communication/dispatcher.rs:45]: oh. If we're checking `is_empty` anyway then, we can replace `drain` with `remove(0)`

Comment (T0mstone) [reply core/editor/src/communication/dispatcher.rs:45]: or wait. This should have used `VecDeque` from the start

**Comment by T0mstone** [comment]
I want to document the traits in `derivable_custom_traits` but I have too little energy right now. I probably won't get around to it soon, so we can move that to a follow-up PR

File: core/editor/src/input/mouse.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+pub struct ViewportPosition {
+	pub x: u32,
+	pub y: u32,
+}
+
+impl ViewportPosition {
+	pub fn distance(&self, other: &Self) -> f64 {
+		let x_diff = other.x as i32 - self.x as i32;
+		let y_diff = other.y as i32 - self.y as i32;
+		f64::sqrt((x_diff * x_diff + y_diff * y_diff) as f64)
```
Comment (T0mstone) [inline core/editor/src/input/mouse.rs:14]: This needs `f64::hypot`

Comment (TrueDoctor) [reply core/editor/src/input/mouse.rs:14]: in that case you have to do two conversions to f64

Comment (TrueDoctor) [reply core/editor/src/input/mouse.rs:14]: You have to do two conversions to f64 but I am not qualified enough to assess the impact that has on each plattform
``` 
        mov     eax, edi
        cvtsi2sd        xmm0, rax
        movapd  xmm1, xmm0
        jmp     qword ptr [rip + hypot@GOTPCREL]
  ```
  vs
  ```
  example::square:
        imul    edi, edi
        imul    esi, esi
        add     esi, edi
        cvtsi2sd        xmm0, rsi
        sqrtsd  xmm0, xmm0
        ret
      ```

Comment (TrueDoctor) [reply core/editor/src/input/mouse.rs:14]: I think hpot might actually be slower: https://github.com/rust-lang/rust/issues/41908
Given we can expect to have very tame inputs I don't actually think hpot would provide that much benefit. But as always micro optimisations without benchmarking are kinda pointless

**Comment by TrueDoctor** [comment]
The only thing left is to translate all the tools. Finish the gray action flow and implement a prototype of the input mapper.
As this pr is already big enough I think it might be sensible to reopen this for review.

**Review by TrueDoctor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)* status: 2 of 67 files reviewed, 14 unresolved discussions (waiting on @pkupper and @T0mstone)

___
*[are_we_proc_macro, line 2 at r3](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_kyy9v1Rthtjl9YG5g-r3-2:-MaJ014M8qBN_J8u5aHW:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/f67f4320bec94b877b34d8aa8b49d0dac87cc68a/are_we_proc_macro#L2)):*
<details><summary><i>Previously, Keavon (Keavon Chambers) wrote…</i></summary><blockquote>

😄 
</blockquote></details>

Done.
___
*[core/document/src/document.rs, line 187 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M__n7F9DfTiSuHHYK4I:-MaJ0YOf2E4QsxJ-JNck:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/document/src/document.rs#L187)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Nitpick: Consistent use of whitespace before match arm return value
</blockquote></details>

Done.
___
*[core/editor/src/communication/dispatcher.rs, line 45 at r4](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_bUINdWdDLmbctYU-3-r4-45:-MaJ0_t56xGNDYJF1IWi:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/7c78d8679f9ffe182d02002376ab139973691991/core/editor/src/communication/dispatcher.rs#L45)):*
<details><summary><i>Previously, T0mstone (T0mstone) wrote…</i></summary><blockquote>

or wait. This should have used `VecDeque` from the start
</blockquote></details>

Done.
___
*[core/editor/src/dispatcher/document_event_handler.rs, line 25 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M__tWf0AKIqhANIF-tj:-MaJ0yWK4E-4yzDD0hYf:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/document_event_handler.rs#L25)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Replace "them" with "it". There is only one action
</blockquote></details>

Done.
___
*[core/editor/src/dispatcher/document_event_handler.rs, line 69 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_a3z2hEjkDUIHv1xPa:-MaJ3KUX3juTfDT0pbKS:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/document_event_handler.rs#L69)):*
<details><summary><i>Previously, TrueDoctor wrote…</i></summary><blockquote>

That would be less efficient because you would have to iterate the vec twice.
This code is pretty much adopted form the filter_drain proposal
</blockquote></details>

Done.
___
*[core/editor/src/dispatcher/global_event_handler.rs, line 23 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_a58bT2r2vPwUulHRr:-MaJ3OJwD-YpZJEsq1JF:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/global_event_handler.rs#L23)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

The name current_actions seems inaccurate as current actions are stored in the member actions. These seem more like static or global actions.
</blockquote></details>

Done.
___
*[core/editor/src/dispatcher/input_manager.rs, line 11 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_aqGIbCKowaSptQSGD:-MaJ3VS5850qAiG8RZR_:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/input_manager.rs#L11)):*
<details><summary><i>Previously, TrueDoctor wrote…</i></summary><blockquote>

adjective: pressed
forced to enlist in the army or navy.

2.
flattened, shaped, or smoothed by the application of pressure, typically from an iron.
"immaculately pressed trousers"

There is a slight semantic difference between expressing that a key is currently in the held down state, vs. the key having been pressed.
If I press a key and let go of it, it is still pressed but no longer depressed
</blockquote></details>

Done.
___
*[core/editor/src/input/mouse.rs, line 14 at r10](https://reviewable.io/reviews/graphiteeditor/graphite/122#-Ma41RFKKleHghyY49wQ-r10-14:-MaJ5oDG2-gX_k-ivgA3:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/c2fbfed899a9829dbc618a9d67777783c5ebbc4e/core/editor/src/input/mouse.rs#L14)):*
<details><summary><i>Previously, TrueDoctor wrote…</i></summary><blockquote>

I think hpot might actually be slower: https://github.com/rust-lang/rust/issues/41908
Given we can expect to have very tame inputs I don't actually think hpot would provide that much benefit. But as always micro optimisations without benchmarking are kinda pointless
</blockquote></details>

Done.
___
*[core/editor/src/document.rs, line 9 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M__pYj16bEjj8g6vwOn:-MaJ3n2c5_8Jf4SI5SWZ:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/document.rs#L9)):*
<details><summary><i>Previously, TrueDoctor wrote…</i></summary><blockquote>

I'm not sure the name is something inherent to the bytes you are saving, I think it would be only associated with the filename which the document library will most likely not handle
</blockquote></details>

Done.
___
*[core/editor/src/macros.rs, line 88 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M__qpNr1RWuzpW9i2fo:-MaJBGfd6VKvD6Y8Wj3m:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/macros.rs#L88)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Document with example (see macros above)
</blockquote></details>

Done.
___
*[core/editor/src/macros.rs, line 95 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M__qyhO7X6kABQS6xj8:-MaJBGzP1C1bGkGozG7j:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/macros.rs#L95)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Document with example
</blockquote></details>

Done.

**Review by TrueDoctor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)* status: 2 of 67 files reviewed, 3 unresolved discussions (waiting on @pkupper and @T0mstone)

___
*[core/editor/src/dispatcher/events.rs, line 40 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/122#-M_aphT8-LLuLH4vSZHf:-MaJEliK0uwCorGIHRGp:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/9f9426a01d0a240db40e2547e1067b087036d792/core/editor/src/dispatcher/events.rs#L40)):*
<details><summary><i>Previously, TrueDoctor wrote…</i></summary><blockquote>

This would be for something like clicking a ui button in the frontend
</blockquote></details>

Done.

**Review by pkupper — CHANGES_REQUESTED** [review]
> Reviewed 6 of 29 files at r2, 4 of 22 files at r3, 1 of 16 files at r4, 1 of 3 files at r5, 1 of 13 files at r7, 2 of 4 files at r9, 23 of 49 files at r10, 2 of 8 files at r11, 3 of 12 files at r12, 10 of 18 files at r13, 7 of 10 files at r14, 5 of 5 files at r15.
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)* status: all files reviewed, 9 unresolved discussions (waiting on @T0mstone and @TrueDoctor)

___
*[client/web/wasm/src/window.rs, line 23 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJLec34Q_f227PCGUF:-MaJLec34Q_f227PCGUG:bnhj21u) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/client/web/wasm/src/window.rs#L23)):*
> ```XML
> }
> 
> /*use editor_core::workspace::PanelId;
> ```

Uncomment or remove
___
*[core/editor/src/input/input_mapper.rs, line 24 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJRw3_2VOyz3WT9KGF:-MaJRw3_2VOyz3WT9KGG:bpbettz) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/editor/src/input/input_mapper.rs#L24)):*
> ```XML
> 
> #[derive(Debug, Clone, Default)]
> struct MappingList(Vec<MappingEntry>);
> ```

Name is confusing because Mappings contains a List of MappingLists. Since this is used to represent an individual key's mapping renaming this to KeyMapping might be better.
___
*[core/editor/src/input/input_mapper.rs, line 41 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJRX2Z5aGW2Ps6W30T:-MaJRX2Z5aGW2Ps6W30U:b3xv0ar) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/editor/src/input/input_mapper.rs#L41)):*
> ```XML
> 
> #[derive(Debug, Clone)]
> struct Mappings {
> ```

Rename to singular "Mapping", since the singular already encompasses a set of MappingEntries. See inconsistent usage in definition of InputMapper.
___
*[core/editor/src/input/keyboard.rs, line 7 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJUZh7BPojc03vIFaq:-MaJUZh7BPojc03vIFar:b9wnsz9) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/editor/src/input/keyboard.rs#L7)):*
> ```XML
> const STORAGE_SIZE: u32 = std::mem::size_of::<usize>() as u32 * 8 + 2 - std::mem::size_of::<StorageType>().leading_zeros();
> const STORAGE_SIZE_BITS: usize = 1 << STORAGE_SIZE;
> const KEY_MASK_STORAGE_LENGHT: usize = (NUMBER_OF_KEYS + STORAGE_SIZE_BITS - 1) >> STORAGE_SIZE;
> ```

Typo: LENGTH
___
*[core/proc-macro/src/lib.rs, line 22 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJh9hC5rRvmpaNQCL3:-MaJh9hC5rRvmpaNQCL4:b-r6dvsn) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/proc-macro/src/lib.rs#L22)):*
> ```XML
> /// This derive macro is enum-only.
> ///
> /// The discriminant enum is a copy of the input enum with all fields of every* variant removed.\
> ```

Remove asterisk
___
*[core/editor/src/document/document_message_handler.rs, line 10 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJNgVfA4263a6DS7QY:-MaJNgVfA4263a6DS7QZ:b3c8jv0) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/editor/src/document/document_message_handler.rs#L10)):*
> ```XML
> #[derive(PartialEq, Clone, Debug)]
> pub enum DocumentMessage {
> 	Operation(DocumentOperation),
> ```

All other messages start with a verb. Maybe something like HandleOperation would be more descriptive?

**Review by TrueDoctor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)* status: 61 of 67 files reviewed, 9 unresolved discussions (waiting on @pkupper and @T0mstone)

___
*[client/web/wasm/src/window.rs, line 23 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJLec34Q_f227PCGUF:-MaJm8KfAtaBpPLB-g0_:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/client/web/wasm/src/window.rs#L23)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Uncomment or remove
</blockquote></details>

Done.
___
*[core/editor/src/input/input_mapper.rs, line 24 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJRw3_2VOyz3WT9KGF:-MaJmwwvB37asdkK3pHL:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/editor/src/input/input_mapper.rs#L24)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Name is confusing because Mappings contains a List of MappingLists. Since this is used to represent an individual key's mapping renaming this to KeyMapping might be better.
</blockquote></details>

Done.
___
*[core/editor/src/input/input_mapper.rs, line 41 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJRX2Z5aGW2Ps6W30T:-MaJn33UDy0JJOl-cuVG:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/editor/src/input/input_mapper.rs#L41)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Rename to singular "Mapping", since the singular already encompasses a set of MappingEntries. See inconsistent usage in definition of InputMapper.
</blockquote></details>

Done.
___
*[core/editor/src/input/keyboard.rs, line 7 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJUZh7BPojc03vIFaq:-MaJn9ZL1dgxQPIYNcba:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/editor/src/input/keyboard.rs#L7)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Typo: LENGTH
</blockquote></details>

Done.
___
*[core/proc-macro/src/lib.rs, line 22 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJh9hC5rRvmpaNQCL3:-MaJnVl95gF7qxZfSTol:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/proc-macro/src/lib.rs#L22)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

Remove asterisk
</blockquote></details>

Done.
___
*[core/editor/src/document/document_message_handler.rs, line 10 at r15](https://reviewable.io/reviews/graphiteeditor/graphite/122#-MaJNgVfA4263a6DS7QY:-MaJmZ5tAUHZNgh5zstS:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/210763425f7c5007c13a40eaa8c8645a850a86d9/core/editor/src/document/document_message_handler.rs#L10)):*
<details><summary><i>Previously, pkupper (Paul Kupper) wrote…</i></summary><blockquote>

All other messages start with a verb. Maybe something like HandleOperation would be more descriptive?
</blockquote></details>

Done.

**Review by pkupper — APPROVED** [review]

**Review by T0mstone — APPROVED** [review]
> Reviewed 6 of 6 files at r16.
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)* status: all files reviewed, 6 unresolved discussions (waiting on @pkupper)

**Review by pkupper — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @TrueDoctor)

**Review by Keavon — APPROVED** [review]

**Review by pkupper — APPROVED** [review]
> Reviewed 18 of 18 files at r17.
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/122)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @TrueDoctor)

**Comment by Keavon** [comment]
🥳 🦜

**Pull request merged**